File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.js
/**
 * @param {number[]} happiness
 * @param {number} k
 * @return {number}
 */
var maximumHappinessSum = function(happiness, k) {
    happiness.sort((a, b) => b - a);
    let result = 0;
    
    for(let counter = 0; counter < k; counter++){
        result += Math.max(0,happiness[counter] - counter);
    }
    
    return result;
};

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        int balancer = 0;
        long result = 0;  
        Arrays.sort(happiness);
        
        while(balancer < k) {
            if(happiness[happiness.length - balancer - 1] - balancer > 0) {
                result += happiness[happiness.length - balancer - 1] - balancer;
                balancer++;
            } else {
                return result;
            }
        }

        return result;
    }
}